packages/viewer/src/lib/scenes/queries.ts

```diff
@@ -94,6 +100,10 @@ export class RootQuery implements ItemQuery<SingleQuery> {
   public withMetadata(filter: string, keys: string[]): MetadataQuery {
     return new MetadataQuery(filter, keys);
   }
+
+  public withSelected(color?: ColorMaterial): AllSelectedQuery {
```

Comment:
Accepting a `color` here is mixing the concept of queries and operations.

Reply:
Not really - the user could select multiple items with different materials - This is a conditional paramter that says, only show the selected items with this color.